### ISSUE TITLE
ps x doesn't work in an alias 

### ISSUE THREAD

**Issue by wreed4** [xonsh/xonsh#1505]
I'm really not sure about this error, but it seems that when I have a function like this

``` python
def _test(args, stdin=None):
    ps x | grep -v grep | grep test
aliases['test'] = _test
```

I get an error like this

```
Signal 18 (CONT) caught by ps (procps version 3.2.8).
```

with no other output.  This often happens just typing `ps x` on the command line as well, but I updated to v0.4.4 and that seems to have fixed that.  But it still doesn't work in an alias.  I even made it a `@foreground` alias even though I'm not sure how that would help.. 

Any thoughts?

EDIT: Actually, `ps x | grep -v test` doesn't work on the command line either.  It does work fine in bash.

**Comment by scopatz** [comment]
Hi @wreed4, thanks for reporting! `ps x | grep -v test` seems to work just fine for me.  Can you tell us a bit more about your system.  What is the output of your `xonfig`?

**Comment by laerus** [comment]
i can reproduce this on debian-stable on current master

```
Signal 18 (CONT) caught by ps (procps-ng version 3.3.9).
/bin/ps:display.c:66: please report this bug
  PID TTY      STAT   TIME COMMAND
...
```
then the normal ouput of ps.

**Comment by wreed4** [comment]
I don't get the normal output afterwards.  Though some googling did bring up that zsh has this problem on debian as well.  

my xonfig is 
```
+------------------+---------------------------------------------------+
| xonsh            | 0.4.4                                             |
| Git SHA          | None                                              |
| Python           | 3.5.1                                             |
| PLY              | 3.7                                               |
| have readline    | True                                              |
| prompt toolkit   | 1.0.3                                             |
| shell type       | prompt_toolkit                                    |
| pygments         | 2.1.3                                             |
| on posix         | <xonsh.lazyasd.LazyBool object at 0x7fc987b889e8> |
| on linux         | True                                              |
| distro           | Red Hat Enterprise Linux Server                   |
| on darwin        | False                                             |
| on windows       | False                                             |
| on cygwin        | False                                             |
| is superuser     | False                                             |
| default encoding | utf-8                                             |
+------------------+---------------------------------------------------+
```

Also, you may want a better `__str__` for LazyBool :P.  I just noticed that.  

Again, this command works fine for me on bash.  Normally I would totally blame the system (I'm at work and we have an old setup and I've messed a lot with it to get it to work the way I want).  But this works in the normal bash prompt.  Even launching bash from xonsh.  Also, in both shells, `which ps` gives `/bin/ps`.

**Comment by scopatz** [comment]
> Also, you may want a better __str__ for LazyBool :P. I just noticed that.

Yeah, that is fixed on master :)

Also, I really wonder what is going on here.  It probably has something to do with signals and job control that we are totally missing.

**Comment by adqm** [comment]
I can't reproduce this on current master (Debian stable).  Is this still an issue, or has this already been taken care of?

**Comment by laerus** [comment]
i can reproduce this anymore either, can't recall of a PR targeting this specifically, we can close for now an reopen if this still persists.

**Comment by wreed4** [comment]
This is not fixed in v0.4.5 that was released 3 days ago.  I just updated and can still replicate.

**Comment by laerus** [comment]
thanks for the info @wreed4! reopened this

**Comment by adqm** [comment]
@wreed4: when you replicated this, was it on 0.4.5 or on the current master?  it's possible it somehow got fixed between 0.4.5 and now...

**Comment by wreed4** [comment]
this was on 0.4.5.  I use it at work so I use the stable versions.  But I can launch a virtualenv and test it out at some point with head.

**Comment by gforsyth** [comment]
Hey @wreed4 -- is this still an issue on 0.4.6?

**Comment by wreed4** [comment]
@gforsyth yes it is.

**Comment by scopatz** [comment]
Ok, I believe that this has now been addressed in the big subprocess refactor in the 0.5.x series.  All of the examples worked for me with out throwing any errors.  Would you mind testing them out @wreed4 and/or closing this issue, please?

**Comment by wreed4** [comment]
I just confirmed that this is no longer an issue for me :)   Thanks for keeping me updated.
